h2o-r/h2o-package/R/frame.R

```diff
@@ -3228,34 +3228,53 @@ as.h2o.Matrix <- function(x, destination_frame="", ...) {
   .key.validate(destination_frame)
   if ( destination_frame=="" ) # .h2o.readSVMLight wont handle ""
     destination_frame <- .key.make("Matrix") # only used if `x` variable name not valid key
-  
-  tmpf <- tempfile(fileext = ".svm")
-  .h2o.write.matrix.svmlight(x, file = tmpf)
-  h2f <- .h2o.readSVMLight(tmpf, destination_frame = destination_frame)
-  file.remove(tmpf)
-  h2f
+
+  drs = as.simple_triplet_matrix(x)# need to convert sparse matrix x to a simple triplet matrix format
+  thefile <- tempfile()
+  .h2o.write_stm_svm(drs, file = thefile)
+  h2f <- h2o.uploadFile(thefile, parse_type = "SVMLight")
+  unlink(thefile)
+  h2f[- c(1)]# remove the first column
```

Comment:
Because he added an extra column in his paste statement to make sure the number of rows of the new frame is the same as the old one.  As a result, I need to remove it before passing things back.